pkg/serverless/invocationlifecycle/trace_test.go

```diff
@@ -7,27 +7,48 @@ package invocationlifecycle
 
 import (
 	"os"
+	"strconv"
 	"testing"
 	"time"
 
 	"github.com/DataDog/datadog-agent/pkg/trace/api"
 	"github.com/stretchr/testify/assert"
 )
 
-func TestStartExecutionSpan(t *testing.T) {
+func TestStartExecutionSpanWithoutPayload(t *testing.T) {
 	startTime := time.Now()
-	startExecutionSpan(startTime)
-
+	startExecutionSpan(startTime, "")
 	assert.Equal(t, startTime, currentExecutionInfo.startTime)
 	assert.NotEqual(t, 0, currentExecutionInfo.traceID)
 	assert.NotEqual(t, 0, currentExecutionInfo.spanID)
 }
+
+func TestStartExecutionSpanWithPayload(t *testing.T) {
+	testString := `a5a{"resource":"/users/create","path":"/users/create","httpMethod":"GET","headers":{"Accept":"*/*","Accept-Encoding":"gzip","x-datadog-parent-id":"1480558859903409531","x-datadog-sampling-priority":"1","x-datadog-trace-id":"5736943178450432258"}}0`
+	startTime := time.Now()
+	startExecutionSpan(startTime, testString)
+	assert.Equal(t, startTime, currentExecutionInfo.startTime)
+	assert.Equal(t, uint64(5736943178450432258), currentExecutionInfo.traceID)
+	assert.Equal(t, uint64(1480558859903409531), currentExecutionInfo.parentID)
+	assert.NotEqual(t, 0, currentExecutionInfo.spanID)
+}
+
+func TestStartExecutionSpanWithPayloadAndInvalidIDs(t *testing.T) {
+	invalidTestString := `a5a{"resource":"/users/create","path":"/users/create","httpMethod":"GET","headers":{"Accept":"*/*","Accept-Encoding":"gzip","x-datadog-parent-id":"INVALID","x-datadog-sampling-priority":"1","x-datadog-trace-id":"INVALID"}}0`
+	startTime := time.Now()
+	startExecutionSpan(startTime, invalidTestString)
```

Comment:
Inside `startExecutionSpan` I set it to reset the IDs before any other logic. It resolved the out of order testing issue I was experiencing before.

Reply:
It might not be causing a problem right now, but I would still suggest doing a cleanup at the end of each test. The reason being that the last test will still leave the global variable in a non-default state. In the future this could cause an issue, so I would say better safe than sorry. Ideally we don't want a test to have any side effects that could "leak" out into other tests.